src/test/java/com/checkmarx/flow/cucumber/integration/end2end/genericendtoend/Repository.java

```diff
@@ -669,39 +661,186 @@ void verifyPRUpdated(String engine) {
+        private Integer webhookId = null;
+
+
+
+        @Override
+        protected void init(GenericEndToEndSteps genericEndToEndSteps) {
+            gitLabProperties = genericEndToEndSteps.gitLabProperties;
+            projectId = getProjectId();
+        }
+
+        @Override
+        Boolean hasWebHook() {
+            log.info("checking if webhook exist in project {}", projectId);
+            HttpEntity<String> httpEntity = new HttpEntity<>(getHeaders());
+            String hooksUrl = String.format("%s%s", gitLabProperties.getApiUrl(), GET_HOOKS_URL);
+            ResponseEntity<String> response = restTemplate.exchange(hooksUrl, HttpMethod.GET, httpEntity, String.class, projectId);
+            JSONArray jsonArray = new JSONArray(response.getBody());
+
+            return !jsonArray.isEmpty();
+        }
+
+        private int getProjectId()
+        {
+            String projectName = repo;
+            String getProjectsUrl = String.format("%s%s%s", gitLabProperties.getApiUrl(), GET_PROJECT_URL, projectName);
+            HttpEntity<String> httpEntity = new HttpEntity<>(getHeaders());
+            ResponseEntity<String> response = restTemplate.exchange(getProjectsUrl, HttpMethod.GET, httpEntity, String.class);
+
+            JSONArray jsonArray = new JSONArray(response.getBody());
+            JSONObject obj = jsonArray.getJSONObject(0);
+
+            int responseProjectId = obj.getInt("id");
+            log.info("found project Id: '{}' for project '{}'", responseProjectId, projectName);
+            return  obj.getInt("id");
+        }
+
+        @Override
+        void generateHook(HookType hookType) {
+
+            log.info("creating new webhook to project '{}'", projectId);
+            String hooksUrl = String.format("%s%s", gitLabProperties.getApiUrl(), CREATE_WEBHOOK_URL);
+            HttpEntity<String> httpEntity = new HttpEntity<>(getHeaders());
+            ResponseEntity<String> response = restTemplate.exchange(hooksUrl, HttpMethod.POST, httpEntity, String.class, projectId, hookTargetURL, gitLabProperties.getWebhookToken());
+            JSONObject obj = new JSONObject(response.getBody());
+            webhookId = obj.getInt("id");
+            log.info("webhook created with Id: '{}'", webhookId);
+        }
+
+        private void pushFileToBranch(String textContent, String path){
+            log.info("pushing file '{}' tp project '{}' ", path, projectId);
+
+            JSONObject jsonObject = new JSONObject();
+            jsonObject.put("branch", "master");
+            jsonObject.put("commit_message", "pushing file");
+            JSONArray jsonArray = new JSONArray();
+            JSONObject actionObject = new JSONObject();
+            actionObject.put("action", "create");
+            actionObject.put("file_path", path);
+            actionObject.put("content", textContent);
+            jsonArray.put(actionObject);
+            jsonObject.put("actions", jsonArray);
```

Comment:
In some cases it makes sense to define a string with the full JSON and then parse it.